### PR TITLE
Make archive-mirrors modifiable via opam config

### LINKED ISSUE #5316: question / feature request: an opam command to set `archive-mirrors` field
Of course, I can manually edit `~/.opam/config`, but it would be great (avoiding typos, etc.) if I could use `opam config set jobs 10` (or `opam config set archive-mirrors http://my-local-mirror`) and also `unset` these again.

As far as I can tell, the `opam config set` and `opam config set-global` are only about variables, not about the data stored in the config file.

### PR DESCRIPTION
**Pull request by hannesm**
fixes #5316

### CODE REVIEW

File: src/client/opamConfigCommand.ml [hunk 1]
```diff
@@ -673,6 +673,9 @@ let global_allowed_fields, global_allowed_sections =
         "jobs", Atomic,
         Config.with_jobs_opt
           (InitConfig.jobs in_config ++ Config.jobs Config.empty);
+        "archive-mirrors", Atomic,
```
Comment (rjbou) [inline src/client/opamConfigCommand.ml:676]: This field is a list of urls, it should be `Modifiable` (appending urls) not `Atomic` (replacement)

**Review by rjbou — CHANGES_REQUESTED** [review]

**Review by rjbou — APPROVED** [review]
Updated to make it an extendable field + add some test

**Comment by hannesm** [comment]
thanks :)

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**